crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -108,13 +111,18 @@ impl AuthenticatorTrait for ZkLoginAuthenticator {
         T: Serialize,
     {
         if aux_verify_data.verify_legacy_zklogin_address {
```

Comment:
this IF is a bit confusing, maybe let's do 

if author != SuiAddress::try_from_padded(&self.inputs)? {
  if !aux_verify_data.verify_legacy_zklogin_address || author != SuiAddress::try_from_unpadded(&self.inputs)? {
    return Err(SuiError::InvalidAddress);
  }
}